### PR TITLE
Fix: PowerBI .pbit parser fails on multiline DAX expressions

### LINKED ISSUE #25483: powerBI .pbit file parser fail on multiline DAX expressions
**Affected module**
Ingestion framework

**Describe the bug**
If a .pbit file has multiline DAX, the parsing fails

**To Reproduce**

create a report with multi-line DAX, export to template (.pbit), ingest with om

**Expected behavior**
should parse

**Version:**
 - OS: ubuntu 24.04 LTS
 - OpenMetadata version: 1.11.16
 - OpenMetadata Ingestion package version: 1.11.16

**Additional context**
Here are some logs extract:

```
[2026-01-23T11:15:33.329759Z] ERROR - file_client.py:136 - Error reading and mapping the datamodel schema file for /tmp/pbitFiles/extracted/Monthly Financial_trusted/Connections: 73 validation errors for PowerBiTable 
measures.0.expression 
  Input should be a valid string [type=string_type, input_value=['', 'VAR PL_Net_Selectio.../ 1000000', ')', '', ''], input_type=list] 
    For further information visit https://errors.pydantic.dev/2.11/v/string_type 
measures.3.expression 
  Input should be a valid string [type=string_type, input_value=['', 'IF(', "    HASONEVA...eference Translation])'], input_type=list] 
    For further information visit https://errors.pydantic.dev/2.11/v/string_type 
measures.4.expression 
  Input should be a valid string [type=string_type, input_value=['', 'IF(', "    HASONEVA...election Translation])'], input_type=list] 
    For further information visit https://errors.pydantic.dev/2.11/v/string_type 
measures.5.expression 
  Input should be a valid string [type=string_type, input_value=['', 'CALCULATE(', "    S...DB P&L'[Source Name]))"], input_type=list] 
    For further information visit https://errors.pydantic.dev/2.11/v/string_type 
measures.6.expression 
  Input should be a valid string [type=string_type, input_value=['', 'SWITCH(TRUE(),', " ...'', '    "Net result")'], input_type=list] 
    For further information visit https://errors.pydantic.dev/2.11/v/string_type 
measures.7.expression 
  Input should be a valid string [type=string_type, input_value=['', "SUMX('DB P&L',[Valu...('Map Account'[Sign]))"], input_type=list] 
    For further information visit https://errors.pydantic.dev/2.11/v/string_type 
measures.8.expression 
  Input should be a valid string [type=string_type, input_value=['', 'VAR PL_Net_Referenc...nce / 1000000', ')', ''], input_type=list] 
    For further information visit https://errors.pydantic.dev/2.11/v/string_type 
```

### PR DESCRIPTION
**Pull request by AntoineGlacet**
## Summary
Fixes #25483 - PowerBI .pbit file ingestion fails when DAX expressions span multiple lines.

## Problem
The PowerBI ingestion connector expected `expression` fields to be strings, but .pbit files store multiline DAX as JSON arrays. This caused Pydantic validation errors during parsing:
```
validation error for PowerBiTable
measures.0.expression
  Input should be a valid string [type=string_type, input_value=['', 'VAR PL_Net_Selectio.../ 1000000', ')', '', ''], input_type=list]
```

## Root Cause
Three Pydantic models had strict `str` typing for expression fields:
- `PowerBiMeasures.expression`
- `PowerBITableSource.expression`  
- `DatasetExpression.expression`

When .pbit files contained multiline DAX, the JSON parser returned lists instead of strings, causing validation failures.

## Solution
1. **Updated type definitions** to accept both `str` and `List[str]`:
   ```python
   expression: Optional[Union[str, List[str]]] = None
   ```

2. **Added Pydantic validators** to normalize lists to newline-joined strings:
   ```python
   @field_validator("expression")
   @classmethod
   def normalize_expression(cls, v):
       return "\n".join(v) if isinstance(v, list) else v
   ```

3. **Added defensive null check** in `metadata.py` to prevent errors when expression is None.

## Changes
- **ingestion/src/metadata/ingestion/source/dashboard/powerbi/models.py** (+31, -5)
  - Updated 3 model expression fields to accept Union[str, List[str]]
  - Added 3 field validators for automatic normalization
- **ingestion/src/metadata/ingestion/source/dashboard/powerbi/metadata.py** (+6, -1)
  - Added null check before processing expressions
- **ingestion/tests/unit/topology/dashboard/test_powerbi_table_measures.py** (+85)
  - Added integration test with real multiline DAX
  - Added 9 unit tests covering all edge cases

### Unit Tests
```bash
pytest tests/unit/topology/dashboard/test_powerbi_table_measures.py -v
# ✅ 13/13 tests passed
```

### Docker Integration Testing
Tested with production .pbit file (Monthly Financial_trusted.pbit):
- **72 multiline DAX measures** - ✅ All parsed successfully
- **32 multiline source expressions** - ✅ All normalized correctly  
- **0 Pydantic validation errors** - ✅ No failures

### Test Coverage
- ✅ String expressions (backward compatibility)
- ✅ List expressions (new behavior)
- ✅ Empty lists (edge case)
- ✅ None values (edge case)
- ✅ All three affected models tested

## Backward Compatibility
✅ **100% backward compatible** - String expressions continue working exactly as before. The Union type and validators only activate when lists are encountered.

## Checklist
- [x] Unit tests added and passing
- [x] Integration tested with real .pbit files
- [x] Follows Pydantic v2 best practices
- [x] No breaking changes
- [x] Defensive coding (null checks)
- [x] Consistent with existing code patterns

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/dashboard/powerbi/models.py [hunk 1]
```diff
@@ -135,17 +135,31 @@ class PowerBiMeasures(BaseModel):
     """
 
     name: str
-    expression: str
+    expression: Optional[Union[str, List[str]]] = None
     description: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (AntoineGlacet) [reply ingestion/src/metadata/ingestion/source/dashboard/powerbi/models.py:139]: This is an intentional change as some measures can have no expressions and it frequently happen in real reports, used in business.

**Comment by AntoineGlacet** [comment]
Hi! I’ve rebased, but main is still moving. I can rebase again if needed, or feel free to resolve any conflicts and merge when convenient.

File: ingestion/src/metadata/ingestion/source/dashboard/powerbi/models.py [hunk 2]
```diff
@@ -135,17 +135,31 @@ class PowerBiMeasures(BaseModel):
     """
 
     name: str
-    expression: str
+    expression: Optional[Union[str, List[str]]] = None
```
Comment (harshsoni2024) [inline ingestion/src/metadata/ingestion/source/dashboard/powerbi/models.py:138]: @AntoineGlacet , looking at the docs we're following from where this model is parsed,
https://learn.microsoft.com/en-us/rest/api/power-bi/admin/workspace-info-get-scan-result#measure
the value of expression in some cases can be List instead of only String as suggested in docs, is that correct ?

File: ingestion/src/metadata/ingestion/source/dashboard/powerbi/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (harshsoni2024) [reply ingestion/src/metadata/ingestion/source/dashboard/powerbi/models.py:139]: makes sense
